packages/confirmation-modal/src/ConfirmationModal/ConfirmationModal.types.ts

```diff
@@ -7,6 +8,14 @@ export const Variant = {
 
 export type Variant = (typeof Variant)[keyof typeof Variant];
 
+interface CustomButtonOnClick {
+  onClick?: () => void;
+}
+type CustomConfirmButtonProps = Omit<ButtonProps, 'variant' | 'onClick'> &
+  CustomButtonOnClick;
+type CustomCancelButtonProps = Omit<ButtonProps, 'onClick'> &
+  CustomButtonOnClick;
```

Comment:
Same question about housing this in the button package. The types are different but maybe it can even be a util that uses `lodash/omit`to narrow the options? It would help with pattern enforcement and make it simpler to identify where custom buttons are being used in the various packages

Reply:
I think this should be fine. we can revisit if it comes up again in the future